### PR TITLE
Partial Fixed: Mobile responsiveness of find and replace dialog box

### LINKED ISSUE #2839: Find and Replace dialog box is not mobile-responsive
### p5.js version

v1.9.0

### What is your operating system?

Mac OS

### Web browser and version

Google Chrome 120.0.6099.129

### Actual Behavior

The dialog box hides behind the editor and hence user is unable to click on the input fields and buttons.

https://github.com/processing/p5.js-web-editor/assets/53830546/bd1afd1f-d76f-4484-afb7-e87b78c7c124

### Expected Behavior

The dialog box should appear above the editor.

### Steps:
1. Open p5.js Web Editor in mobile screen
2. In the Edit menu, click Replace option
3. Find and Replace dialog box will appear

### PR DESCRIPTION
**Pull request by mrkirthi-24**
Partial Fixes #2839 

Changes:

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] has no test errors (`npm run test`)
* [x] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: client/styles/components/_editor.scss [hunk 1]
```diff
@@ -79,6 +79,7 @@ pre.CodeMirror-line {
   }
   // left: 0 !important;
   width: #{48 / $base-font-size}rem;
+  z-index: 0;
```
Comment (lindapaiste) [inline client/styles/components/_editor.scss]: We probably don't need an explicit z-index here.

It looks like there's a `z-index: 4` on the `.CodeMirror-gutter-wrapper` div which is coming from the core CodeMirror library.  I assume it's there for a reason so let's not mess with it.

File: client/styles/components/_editor.scss [hunk 2]
```diff
@@ -96,6 +97,7 @@ pre.CodeMirror-line {
     right: 0;
     width: 100%;
     margin-left: 0;
+    z-index: 5;
   }
 
   z-index: 1;
```
Comment (lindapaiste) [inline client/styles/components/_editor.scss]: Let's change the existing `z-index: 1` (line 101) to `z-index: 5`, regardless of breakpoint.  Like it's not an issue on larger screens because the dialog doesn't overlap the gutter, but it does seem "right" that it should have a higher z-index.

File: client/styles/components/_editor.scss [hunk 3]
```diff
@@ -139,6 +141,10 @@ pre.CodeMirror-line {
 .CodeMirror-find-controls {
   display: flex;
   align-items: center;
+
+  @media (max-width: 580px) {
+    padding: 0.6rem 0rem;
+  }
```
Comment (lindapaiste) [inline client/styles/components/_editor.scss]: This seems unrelated to the `z-index` issues.  So let's save that for a PR where we fix the layout.

Comment (mrkirthi-24) [reply client/styles/components/_editor.scss]: OK, I will raise a new pr for it. Thank you

**Comment by lindapaiste** [comment]
I'm sorry but I forgot about this PR!  I'm going to close it because the `z-index` was addressed alongside other issues in #3037
